pkg/container/client.go

```diff
@@ -343,6 +358,26 @@ func (client dockerClient) HasNewImage(ctx context.Context, container t.Containe
 		return false, currentImageID, nil
 	}
 
+	// Disabled by default
+	if params.DelayDays > 0 {
+		// Define the layout string for the date format without milliseconds
+		layout := "2006-01-02T15:04:05Z"
+		newImageDate, error := time.Parse(layout, truncateMilliseconds(newImageInfo.Created))
+
+		if error != nil {
+			log.Errorf("Error parsing Created date (%s) for container %s latest label. Error: %s", newImageInfo.Created, container.Name(), error)
+			return false, currentImageID, nil
+		} else {
+			requiredDays := params.DelayDays
+			diffDays := int(time.Since(newImageDate).Hours() / 24)
+
+			if diffDays < requiredDays {
+				log.Infof("New image found for %s that was created %d day(s) ago but update delayed until %d day(s) after creation", container.Name(), diffDays, requiredDays)
+				return false, currentImageID, nil
+			}
+		}
+	}
```

Comment:
I think this check should be in either `IsContainerStale` (not marking the container as stale if it's being delayed which maybe isn't technically correct), or in `internal/actions/update.go`, adding it to the "shouldUpdate" logic, which seems like the most expected place.

https://github.com/containrrr/watchtower/blob/0a14f3aa9c7fff2e69f5c3796f24ba9172d25d0e/internal/actions/update.go#L37

(the method logic could still be in `client.go`, but it would be called from `update.go`)

I guess the reason for putting it here is that you already have the image info struct, but actually, re-fetching it with `ImageInspectWithRaw` in another method adds only a negligible overhead and allows for much clearer separation of code.

Reply:
Oops I meant to add my comment here rather than as a new one. Just adding a note here to indicate that I believe the comment is addressed in the latest commits. Thanks.